#! /bin/sh /usr/share/dpatch/dpatch-run
## 62_warn_on_missing_required_fields.dpatch by Kel Modderman <kel@otaku42.de>
##
## DP: Goal: Emit warning when script contains header with missing Provides,
## DP: Required-Start or Required-Stop LSB comment keywords. This is
## DP: consistent with lintian warning:
## DP:   http://lintian.debian.org/reports/tags/init.d-script-missing-lsb-keyword.html
## DP:
## DP: Fixes: none
## DP:
## DP: Status: Intend to submit upstream.
## DP:
## DP: Notes: test_broken_header test suite case exposes this situation.

@DPATCH@
diff -urNad insserv~/insserv.c insserv/insserv.c
--- insserv~/insserv.c	2008-05-19 23:36:39.000000000 +1000
+++ insserv/insserv.c	2008-05-19 23:38:11.214013731 +1000
@@ -1257,9 +1257,9 @@
     }
 
 #ifndef USE_STOP_TAGS
-    if (verbose && (begin && end && (!provides || !required_start)))
+    if (begin && end && (!provides || !required_start))
 #else  /* USE_STOP_TAGS */
-    if (verbose && (begin && end && (!provides || !required_start || !required_stop)))
+    if (begin && end && (!provides || !required_start || !required_stop))
 #endif /* USE_STOP_TAGS */
     {
 	char *name = basename(path);
